gold.mixin

```diff
@@ -0,0 +1,11 @@
+{
+    "build": {
+        "gold": {
+            "cmake-args": [
+                "-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=gold",
+                "-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=gold",
+                "-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=gold"
```

Comment:
```suggestion
                "-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=gold",
                "-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=gold",
                "-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=gold",
                "-DCMAKE_STATIC_LINKER_FLAGS=-fuse-ld=gold"
```

Swapped the order of elements to be lexically sorted and added an entry for [CMAKE_STATIC_LINKER_FLAGS](https://cmake.org/cmake/help/v3.18/variable/CMAKE_STATIC_LINKER_FLAGS.html?highlight=linker_flags#variable:CMAKE_STATIC_LINKER_FLAGS).

Reply:
Presuming the same feedback from https://github.com/colcon/colcon-mixin-repository/pull/28#discussion_r779054700 applies this suggestion is not correct.